src/ochanticipy/datasources/codab/codab.py

```diff
@@ -56,12 +75,16 @@ def download(self, clobber: bool = False) -> Path:
         >>> codab = CodAB(country_config=country_config)
         >>> npl_cod_shapefile = codab.download()
         """
-        return self._download(
-            filepath=self._raw_filepath,
-            hdx_address=f"cod-ab-{self._country_config.iso3}",
-            hdx_dataset_name=self._datasource_config.hdx_dataset_name,
-            clobber=clobber,
-        )
+        for filepath, hdx_resource_name in zip(
+            self._raw_filepaths, self._hdx_resource_list
+        ):
+            self._download(
+                filepath=filepath,
+                hdx_dataset=f"cod-ab-{self._country_config.iso3}",
+                hdx_resource_name=hdx_resource_name,
+                clobber=clobber,
+            )
+        return self._base_dir
```

Comment:
In the case of multiple files, I try to return the full path to the directory containing the files

Reply:
In this case, I think we are doing that. This should return the base directory where all of the zip files are stored. Since we don't unzip any of the files (if multiple), `self._base_dir` should be the directory that contains all of the files (the zip files). Should I change it to be the path to the zip file if a single file is downloaded, and the directory if multiple?

Reply:
[here](https://github.com/OCHA-DAP/ocha-anticipy/pull/193/files#diff-bab90d7c88ef7e9d8c3300c7c71b4733fb7c1312252adefdb2481416ab892991R34) is where you set `base_dir`, it only ends up being `{iso3}_cod_ab`

> Should I change it to be the path to the zip file if a single file is downloaded, and the directory if multiple?

You could but I wouldn't bother if it's too complicated